src/rtc.rs

```diff
@@ -315,10 +315,22 @@ impl Rtc {
         };
         if clear {
             self.write(false, |rtc| match event {
-                Event::WakeupTimer => rtc.isr.modify(|_, w| w.wutf().clear_bit()),
-                Event::AlarmA => rtc.isr.modify(|_, w| w.alraf().clear_bit()),
-                Event::AlarmB => rtc.isr.modify(|_, w| w.alrbf().clear_bit()),
-                Event::Timestamp => rtc.isr.modify(|_, w| w.tsf().clear_bit()),
+                Event::WakeupTimer => {
+                    rtc.isr.modify(|_, w| w.wutf().clear_bit());
+                    unsafe { (*EXTI::ptr()).pr1.write(|w| w.bits(1 << 20)) };
+                }
+                Event::AlarmA => {
+                    rtc.isr.modify(|_, w| w.alraf().clear_bit());
+                    unsafe { (*EXTI::ptr()).pr1.write(|w| w.bits(1 << 18)) };
+                }
+                Event::AlarmB => {
+                    rtc.isr.modify(|_, w| w.alrbf().clear_bit());
+                    unsafe { (*EXTI::ptr()).pr1.write(|w| w.bits(1 << 18)) };
```

Comment:
Are AlarmA and AlarmB on the same EXTI line?

Reply:
Unfortunately yep..

Side note: Would it be okay to enable EXTI stuff in the same way as this (unsafe ptr) in listen/unlisten to avoid the exti argument?

Reply:
In listen and unlisten I would not use ptrs as there one want to understand the relationship to EXTI.
However, if one enables the interrupts (via EXTI) the connection has been made that we are now dependent on EXTI and using ptrs for the clearing of bits is IMO fine.
Else the interface would not be ergonomic. But here there are any conflicting views.

Best is to usually follow the convention in the HAL so the same expectations are held. :)